server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailSet.scala

```diff
@@ -96,12 +117,94 @@ case class EmailCreationRequest(mailboxIds: MailboxIds,
+      attachments
+        .filter(!_.isInline)
+        .map(attachment => getAttachmentMetadata(attachment, attachmentManager, mailboxSession))
+        .map(attachmentMetadataList => attachmentMetadataList
+          .flatMap(attachmentAndMetadata => loadAttachment(attachmentAndMetadata._1, attachmentAndMetadata._2, attachmentContentLoader, mailboxSession)))
+        .sequence
+
+    maybeAttachments.map(list => {
+      list.foldLeft(multipartBuilder) {
+        case (builder, (attachment, attachmentMetadata, content)) =>
+          bodypartBuilder.setBody(content, attachment.`type`.value)
+            .setField(contentTypeField(attachment, attachmentMetadata))
+            .setContentDisposition(attachment.disposition.getOrElse(Disposition.ATTACHMENT).value)
+
+          if (attachment.cid.isDefined) {
+            bodypartBuilder.setField(new RawField("Content-ID", attachment.cid.get))
+          }
+
+          builder.addBodyPart(bodypartBuilder)
+          builder
+      }
     })
+  }
+
+  private def contentTypeField(attachment: Attachment, attachmentMetadata: AttachmentMetadata): ContentTypeField = {
+    val typeAsField: ContentTypeField = attachmentMetadata.getType.asMime4J
+    if (attachment.name.isDefined) {
+      Fields.contentType(typeAsField.getMimeType,
+        Map.newBuilder[String, String]
+          .addAll(parametersWithoutName(typeAsField))
+          .addOne("name", EncoderUtil.encodeEncodedWord(attachment.name.get.value, Usage.TEXT_TOKEN))
+          .result
+          .asJava);
+    }
+    typeAsField;
+  }
+
+  private def parametersWithoutName(typeAsField: ContentTypeField): Map[String, String] = {
+    typeAsField.getParameters
+      .asScala
+      .filter(!_._1.equals("name"))
+      .toMap
+  }
+
+  private def getAttachmentMetadata(attachment: Attachment,
+                                    attachmentManager: AttachmentManager,
+                                    mailboxSession: MailboxSession): Either[AttachmentNotFoundException, (Attachment, AttachmentMetadata)] = {
+    Try(attachmentManager.getAttachment(AttachmentId.from(attachment.blobId.value.toString), mailboxSession))
+      .fold(e => Left(AttachmentNotFoundException(e, s"Attachment not found: ${attachment.blobId.value}")),
+        attachmentMetadata => Right((attachment, attachmentMetadata)))
+  }
+
+  private def loadAttachment(attachment: Attachment,
+                             attachmentMetadata: AttachmentMetadata,
+                             attachmentContentLoader: AttachmentContentLoader,
+                             mailboxSession: MailboxSession): Either[IllegalArgumentException, (Attachment, AttachmentMetadata, Array[Byte])] = {
+    Try(attachmentContentLoader.load(attachmentMetadata, mailboxSession))
+      .toEither
+      .fold(e => Left(AttachmentNotFoundException(e, s"Attachment not found: ${attachment.blobId.value}")),
```

Comment:
Here, attachmentContentLoader.load might throw `` & ``

You are considering IOException as not found.

Maybe a "catch" is more straightforward and less error prone?

```
try {
     val inputStream = attachmentContentLoader.load(attachmentMetadata, mailboxSession)
     scala.Right((attachment, attachmentMetadata, inputStream.readAllBytes()))
 } catch {
    case e: AttachmentNotFoundException => Left(AttachmentNotFoundException(...))
    case e: IOException => ???
}
```

And the upstream method likely need to handle IOException too... (they should fail)